### PR TITLE
Pivotal ID # 187277504: Submission Level Permission

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
https://www.pivotaltracker.com/story/show/187277504

- Fix logic to provide permissions over submissions
- Create an operation to provide submission level permissions
- Update resources and operation names
- Remove unused code

### CODE REVIEW

File: client/bio-commandline/src/main/kotlin/uk/ac/ebi/biostd/client/cli/services/SecurityService.kt [hunk 1]
```diff
@@ -3,14 +3,23 @@ package uk.ac.ebi.biostd.client.cli.services
 import uk.ac.ebi.biostd.client.cli.dto.PermissionRequest
 
 internal class SecurityService {
-    fun grantPermission(request: PermissionRequest) = performRequest { permission(request) }
+    fun grantCollectionPermission(request: PermissionRequest) =
```
Comment (Juan-EBI) [inline client/bio-commandline/src/main/kotlin/uk/ac/ebi/biostd/client/cli/services/SecurityService.kt:6]: wondering if this really is a collection, basically we give permission to access No, like a submission I am wondering if command, documentaiton and this method name should reflect that.

File: client/bio-webclient/src/main/kotlin/ac/uk/ebi/biostd/client/integration/web/SubmitClient.kt [hunk 2]
```diff
@@ -226,9 +226,15 @@ interface ExtSubmissionOperations {
...
 interface PermissionOperations {
-    fun givePermissionToUser(
+    fun grantCollectionPermission(
         user: String,
-        accessTagName: String,
+        accNo: String,
+        accessType: String,
+    )
+
+    fun grantSubmissionPermission(
```
Comment (Juan-EBI) [inline client/bio-webclient/src/main/kotlin/ac/uk/ebi/biostd/client/integration/web/SubmitClient.kt:235]: I think we should not mantain two operations but a single one. grantPermision depending if accNo is a collection or a submission I do not think should be relevant. `userPermissionsService.hasPermission` does not care already

**Review by Juan-EBI — COMMENTED** [review]
Looking good please check comment, it may be subjetive so we can discuss it

**Comment by jhoanmanuelms** [comment]
Closed in favor of #844

**Pull request closed without merging**